src/lib.rs

```diff
@@ -187,14 +208,18 @@ impl Client {
     ///
     /// Will shutdown the connection task as well as the beat task (if running)
     /// If we still have a connection to faktory, this will send the END command as well.
-    pub fn close(self) -> Result<()> {
+    pub async fn close(mut self) -> Result<()> {
```

Comment:
Won't this consume the client? We will have to clone it again here, I think: https://github.com/systeminit/si/blob/main/bin/pinga/src/main.rs#L214

Reply:
Yes it consumes, I don't see how? We already have a client and we are at the end of the loop, we create the client at the beggining.

Reply:
We are also calling close in the `select` above, so won't it be moved before we get to that line? Sorry if I'm misunderstanding!

Reply:
The select above breaks out of the loop so it doesn't hurt. I actually removed that loop in my local changes tho.

Don't worry I'm already using this branch and don't have a problem. If I do I can just change in a following hotfix PR, thanks for double checking tho.